src/array_api_extra/_funcs.py

```diff
@@ -548,3 +561,348 @@ def sinc(x: Array, /, *, xp: ModuleType | None = None) -> Array:
+        """Allow for the alternate syntax ``at(x)[start:stop:step]``,
+        which looks prettier than ``at(x, slice(start, stop, step))``
+        and feels more intuitive coming from the JAX documentation.
+        """
+        if self.idx is not _undef:
+            msg = "Index has already been set"
+            raise ValueError(msg)
+        self.idx = idx
+        return self
+
+    def _check_args(self, /, copy: bool | None) -> None:
+        if self.idx is _undef:
+            msg = (
+                "Index has not been set.\n"
+                "Usage: either\n"
+                "    at(x, idx).set(value)\n"
+                "or\n"
+                "    at(x)[idx].set(value)\n"
+                "(same for all other methods)."
+            )
+            raise TypeError(msg)
+
+        if copy not in (True, False, None):
+            msg = f"copy must be True, False, or None; got {copy!r}"  # pyright: ignore[reportUnreachable]
+            raise ValueError(msg)
+
+    def get(
+        self,
+        /,
+        copy: bool | None = True,
+        xp: ModuleType | None = None,
+    ) -> Array:
+        """Return ``xp.asarray(x[idx])``. In addition to plain ``__getitem__``, this allows
+        ensuring that the output is either a copy or a view; it also allows passing
+        keyword arguments to the backend.
+        """
+        self._check_args(copy=copy)
+        x = self.x
+
+        if copy is False:
+            if is_array_api_obj(self.idx):
+                # Boolean index. Note that the array API spec
+                # https://data-apis.org/array-api/latest/API_specification/indexing.html
+                # does not allow for list, tuple, and tuples of slices plus one or more
+                # one-dimensional array indices, although many backends support them.
+                # So this check will encounter a lot of false negatives in real life,
+                # which can be caught by testing the user code vs. array-api-strict.
+                msg = "get() with an array index always returns a copy"
+                raise ValueError(msg)
+
+            # Prevent scalar indices together with copy=False.
+            # Even if some backends may return a scalar view of the original, we chose to be
+            # strict here beceause some other backends, such as numpy, definitely don't.
+            tup_idx = self.idx if isinstance(self.idx, tuple) else (self.idx,)
+            if any(
+                i is not None and i is not Ellipsis and not isinstance(i, slice)
+                for i in tup_idx
+            ):
+                msg = "get() with a scalar index typically returns a copy"
+                raise ValueError(msg)
+
+            # Note: this is not the same list of backends as is_writeable_array()
+            if is_dask_array(x) or is_jax_array(x) or is_pydata_sparse_array(x):
+                msg = f"get() on {array_namespace(x)} arrays always returns a copy"
+                raise ValueError(msg)
```

Comment:
These checks are very brittle and I'm not comfortable going on with `get()` as it is.

I can see two ways forward:
1. Write a new function in `array-api-compat`, `getitem_returns_view(x: Array, idx: Index) -> bool` that explores all the nooks and crannies of all the possible intersections of arrays and indices, then use it here
2. Remove `get()` altogether. I have a strong suspicion it may not be needed in scipy to begin with.

Reply:
The only use I can think of out of `get(copy=True)` would be to avoid an unnecessary double copy when idx is a boolean mask

Reply:
I've removed get(), at least for the time being.